server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -524,10 +524,11 @@ public void applyClusterState(ClusterChangedEvent event) {
                         logger.warn(() -> new ParameterizedMessage("failed to create repository [{}]", repositoryMetadata.name()), ex);
                     }
                 }
-                if (repository != null) {
-                    logger.debug("registering repository [{}]", repositoryMetadata.name());
-                    builder.put(repositoryMetadata.name(), repository);
+                if (repository == null) {
+                    repository = new DummyRepository(repositoryMetadata);
```

Comment:
I would rather do this within the `catch` blocks above, using the exception that is thrown by the repository creation attempt. At the moment these blocks catch `RepositoryException` but we should instead catch `Exception` and then `assert ex instance RepositoryException` to fail tests on other kinds of exception.

Reply:
I didn't change `RepositoryException` to `Exception` here ,because `RepositoriesService#createRepository` have done such thing and make sure exception is `RepositoryException`.

Reply:
Ah good point